lib/search.js

```diff
@@ -17,28 +17,47 @@ module.exports = new class LibSearch {
     }
 
     optimizeQueryTerms(input) {
+        // parse
+        let normalize = this.normalizeProt();
+        input.imdbid = input.imdbid && parseInt(input.imdbid.toString().replace('tt', ''), 10)
+        input.season = input.season && parseInt(input.season)
+        input.episode = input.episode && parseInt(input.episode)
+        input.hash = input.hash && input.hash.toString().length >= 32 && input.hash.toString().toLowerCase()
+        input.filesize = input.filesize && parseInt(input.filesize)
+        input.fps = input.fps && input.fps.toString()
+        input.filename = input.filename || input.path && path.basename(input.path)
+        input.tags = input.filename && normalize(input.filename).toLowerCase().match(/[a-z0-9]{2,}/gi)
+
         return this.checkHash(input).then(obj => {
             const output = Array()
             let i = 0
 
-            if (obj) { // first data call
-                output[i] = obj
+            if (obj) { 
+                input.hash = obj.moviehash.toLowerCase()
+                input.filesize = parseInt(obj.moviebytesize)
+            }
+            // first data call
+            if (input.hash || input.filesize) {
+                output[i] = {
+                    moviehash: input.hash && input.hash, 
+                    moviebytesize: input.filesize && input.filesize.toString() 
+                }
                 i++
             }
 
-            if (input.filename || input.path) { // second data call
+            if (input.filename) { // second data call
                 output[i] = Object()
-                output[i].tag = input.filename || path.basename(input.path)
+                output[i].tag = input.filename
                 i++
             }
 
             if (input.imdbid) { // third data call
                 output[i] = Object()
-                output[i].imdbid = input.imdbid.toString().replace('tt', '')
+                output[i].imdbid = input.imdbid.toString()
 
                 if (input.season && input.episode) {
-                    output[i].season = parseInt(input.season).toString()
-                    output[i].episode = parseInt(input.episode).toString()
+                    output[i].season = input.season.toString()

```

Comment:
Keep that because I saw that you worried in sending each term as a string, if I delete that the property is going to be sent as an integer, is that okay?